pennylane/compiler/python_compiler/transforms/convert_to_mbqc_formalism.py

```diff
@@ -0,0 +1,770 @@
+
+    def _rotxzx_measurements(
+        self, graph_qubits_dict, op, rewriter: pattern_rewriter.PatternRewriter
+    ):
+        """Insert measurement ops for a RotXZX gate and return measurement results and the result graph qubits"""
+        m1, graph_qubits_dict[1] = self._measure_x_op(graph_qubits_dict[1], op, rewriter)
+        m2, graph_qubits_dict[2] = self._cond_insert_arbitary_basis_measure_op(
+            m1, op.params[0], "XY", graph_qubits_dict[2], op, rewriter
+        )
+        m3, graph_qubits_dict[3] = self._cond_insert_arbitary_basis_measure_op(
+            m2, op.params[1], "XY", graph_qubits_dict[3], op, rewriter
+        )
+        m4, graph_qubits_dict[4] = self._cond_insert_arbitary_basis_measure_op(
+            m3, op.params[2], "XY", graph_qubits_dict[4], op, rewriter
+        )
+        return [m1, m2, m3, m4], graph_qubits_dict
+
+    def _queue_measurements(
+        self, graph_qubits_dict: dict, op: CustomOp, rewriter: pattern_rewriter.PatternRewriter
+    ):
+        """Add mid-measurement ops to the IR and return the measurement results and the updated graph qubit dict.
+        Args:
+            graph_qubits_dict (dict) : A dict stores all qubit info in a graph state.
+            op (CustomOp) : A gate operation object.
+            rewriter (pattern_rewriter.PatternRewriter): A pattern rewriter.
+
+        Returns:
+            A list of mid-measurement results and the updated graph qubit dict.
+        """
+        match op.gate_name.data:
+            case "Hadamard":
+                return self._hadamard_measurements(graph_qubits_dict, op, rewriter)
+            case "S":
+                return self._s_measurements(graph_qubits_dict, op, rewriter)
+            case "RZ":
+                return self._rz_measurements(graph_qubits_dict, op, rewriter)
+            case "RotXZX":
+                return self._rotxzx_measurements(graph_qubits_dict, op, rewriter)
+            case "CNOT":
+                return self._cnot_measurements(graph_qubits_dict, op, rewriter)
+
+    def _insert_cond_byproduct_op(
+        self,
+        parity_res: OpResult,
+        gate_name: str,
+        qubit: QubitType,
+        op: CustomOp,
+        rewriter: pattern_rewriter.PatternRewriter,
+    ):  # pylint: disable=too-many-arguments, too-many-positional-arguments
+        """Insert a byproduct op related operations to the IR.
+        Args:
+            parity_res (OpResult) : Parity check result.
+            gate_name (str) : The name of gate to be corrected.
+            qubit (QubitType) : The result auxiliary qubit to be corrected.
+            op (CustomOp) : A gate operation object.
+            rewriter (pattern_rewriter.PatternRewriter): A pattern rewriter.
+
+        Return:
+            The result auxiliary qubit.
+        """
+        # Create a const op, which hold the integer `1`.
+        constantOneOp = arith.ConstantOp.from_int_and_width(1, builtin.i1)
```

Comment:
Just curious where the `from_int_and_width` is documented? I couldn't find it in the wild on the MLIR docs. Is it xDSL specific?

Reply:
Yes, it's xdsl specific. [Here](https://github.com/xdslproject/xdsl/blob/0c118332260fcec1f66d5547096b4090f1085fe9/xdsl/dialects/arith.py#L158) is where it is defined.